psi4/src/psi4/libsapt_solver/usapt0.cc

```diff
@@ -749,42 +749,42 @@ void USAPT0::fock_terms() {
-    Exch10_2dm_terms[1] += Matrix::triplet(Db_A,S,Matrix::triplet(Db_B,Ja_A,Db_B))->vector_dot(S);
-    Exch10_2dm_terms[1] += Matrix::triplet(Db_A,S,Matrix::triplet(Db_B,V_A,Db_B))->vector_dot(S);
-    Exch10_2dm_terms[1] += Matrix::triplet(Db_A,S,Matrix::triplet(Db_B,S,Db_A))->vector_dot(Jb_B);
-    Exch10_2dm_terms[1] += Matrix::triplet(Db_A,S,Matrix::triplet(Db_B,S,Db_A))->vector_dot(Ja_B);
-    Exch10_2dm_terms[1] += Matrix::triplet(Db_A,S,Matrix::triplet(Db_B,S,Db_A))->vector_dot(V_B);
-    Exch10_2dm_terms[2] += Matrix::triplet(Da_B,S,Da_A)->vector_dot(Ka_B);
-    Exch10_2dm_terms[2] -= Matrix::triplet(Da_B,S,Da_A)->vector_dot(Jb_B);
-    Exch10_2dm_terms[2] -= Matrix::triplet(Da_B,S,Da_A)->vector_dot(Ja_B);
-    Exch10_2dm_terms[2] -= Matrix::triplet(Da_B,S,Da_A)->vector_dot(V_B);
-    Exch10_2dm_terms[3] += Matrix::triplet(Db_B,S,Db_A)->vector_dot(Kb_B);
-    Exch10_2dm_terms[3] -= Matrix::triplet(Db_B,S,Db_A)->vector_dot(Ja_B);
-    Exch10_2dm_terms[3] -= Matrix::triplet(Db_B,S,Db_A)->vector_dot(Jb_B);
-    Exch10_2dm_terms[3] -= Matrix::triplet(Db_B,S,Db_A)->vector_dot(V_B);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,Da_B)->vector_dot(Ka_A);
+    Exch10_2dm_terms[0] -= linalg::triplet(Da_A,S,Da_B)->vector_dot(Ja_A);
+    Exch10_2dm_terms[0] -= linalg::triplet(Da_A,S,Da_B)->vector_dot(Jb_A);
+    Exch10_2dm_terms[0] -= linalg::triplet(Da_A,S,Da_B)->vector_dot(V_A);
+    Exch10_2dm_terms[0] -= linalg::triplet(Da_A,S,Da_B)->vector_dot(Ka_AB);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,Ja_A,Da_B))->vector_dot(S);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,Jb_A,Da_B))->vector_dot(S);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,V_A,Da_B))->vector_dot(S);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,S,Da_A))->vector_dot(Ja_B);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,S,Da_A))->vector_dot(Jb_B);
+    Exch10_2dm_terms[0] += linalg::triplet(Da_A,S,triplet(Da_B,S,Da_A))->vector_dot(V_B);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,Db_B)->vector_dot(Kb_A);
+    Exch10_2dm_terms[1] -= linalg::triplet(Db_A,S,Db_B)->vector_dot(Jb_A);
+    Exch10_2dm_terms[1] -= linalg::triplet(Db_A,S,Db_B)->vector_dot(Ja_A);
+    Exch10_2dm_terms[1] -= linalg::triplet(Db_A,S,Db_B)->vector_dot(V_A);
+    Exch10_2dm_terms[1] -= linalg::triplet(Db_A,S,Db_B)->vector_dot(Kb_AB);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,Jb_A,Db_B))->vector_dot(S);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,Ja_A,Db_B))->vector_dot(S);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,V_A,Db_B))->vector_dot(S);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,S,Db_A))->vector_dot(Jb_B);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,S,Db_A))->vector_dot(Ja_B);
+    Exch10_2dm_terms[1] += linalg::triplet(Db_A,S,triplet(Db_B,S,Db_A))->vector_dot(V_B);
```

Comment:
do the second `triplet`s in the above six lines need `linalg::triplet`?

Reply:
Yes, but this portion of the code is commented out.